signer/src/api/info.rs

```diff
@@ -0,0 +1,505 @@
+    pub rounds: u32,
+    pub target_rounds: u32,
+    pub current_aggregate_key: Option<String>,
+    pub min_bitcoin_block_height: Option<u64>,
+}
+
+impl Default for InfoResponse {
+    fn default() -> Self {
+        // This is a build-time constant and varies per build environment,
+        // so this lint is not applicable.
+        #[allow(clippy::const_is_empty)]
+        let target_env_abi = if crate::TARGET_ENV_ABI.is_empty() {
+            None
+        } else {
+            Some(crate::TARGET_ENV_ABI.to_string())
+        };
+
+        Self {
+            bitcoin: BitcoinInfo {
+                signer_tip: None,
+                node_tip: None,
+                node_chain: None,
+                node_version: None,
+                node_subversion: None,
+            },
+            stacks: StacksInfo {
+                signer_tip: None,
+                node_tip: None,
+                node_bitcoin_block_height: None,
+                node_version: None,
+            },
+            dkg: DkgInfo {
+                rounds: 0,
+                current_aggregate_key: None,
+                target_rounds: 1,
+                min_bitcoin_block_height: None,
+            },
+            build_info: BuildInfo {
+                rust_version: crate::RUSTC_VERSION.to_string(),
+                git_revision: crate::GIT_COMMIT.to_string(),
+                target_arch: crate::TARGET_ARCH.to_string(),
+                target_env_abi,
+            },
+            timestamp: time::OffsetDateTime::now_utc().to_string(),
+        }
+    }
+}
+
+impl IntoResponse for InfoResponse {
+    fn into_response(self) -> axum::response::Response {
+        Json(self).into_response()
+    }
+}
+
+/// Handler for the `/info` endpoint. This method is infallible and returns
+/// `null` for any missing information.
+pub async fn info_handler<C: Context>(state: State<ApiState<C>>) -> InfoResponse {
+    let bitcoin_client = state.ctx.get_bitcoin_client();
+    let stacks_client = state.ctx.get_stacks_client();
+    let storage = state.ctx.get_storage();
+    let config = state.ctx.config();
+
+    let mut response = InfoResponse::default();
+
+    response.populate_local_chain_info(&storage).await;
+    response.populate_bitcoin_node_info(&bitcoin_client).await;
+    response.populate_stacks_node_info(&stacks_client).await;
+    response.populate_dkg_info(&storage, config).await;
+
+    response
+}
+
+impl InfoResponse {
+    /// Populates the local Bitcoin and Stacks chain tip information.
+    async fn populate_local_chain_info(&mut self, storage: &impl DbRead) {
+        match storage.get_bitcoin_canonical_chain_tip().await {
+            Ok(Some(local_bitcoin_chain_tip)) => {
+                let bitcoin_block = storage
+                    .get_bitcoin_block(&local_bitcoin_chain_tip)
+                    .await
+                    .inspect_err(|e| {
+                        tracing::error!("error reading bitcoin block from the database: {}", e)
```

Comment:
I'm wondering if it could be helpful to also have those errors in the response, so that we could get useful info without having to dig into logs.

Reply:
I thought about it, but landed in that we should probably have a separate `/health` endpoint which tests connectivity to the different dependencies and returns errors if encountered.